src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -592,3 +595,70 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
   }
   return i;
 }
+
+/// Check if a character is a digit with respect to the given radix, e.g. if the
+/// radix is 10 then check if the character is in the range 0-9.
+/// \param chr: the character
+/// \param radix:  the radix
+/// \return an expression for the condition
+exprt is_digit_with_radix(exprt chr, exprt radix)
+{
+  const typet &char_type=chr.type();
+  exprt zero_char=from_integer('0', char_type);
+  exprt nine_char=from_integer('9', char_type);
+  exprt a_char=from_integer('a', char_type);
+  exprt A_char=from_integer('A', char_type);
+
+  and_exprt true_case(
```

Comment:
Rename true-case and false-case for what they are, not how they're used-- perhaps in_range_radix_le_10 and _gt_10?